picard/ui/widgets/editablelistview.py

```diff
@@ -134,6 +134,47 @@ def _move_rows_relative(self, rows, direction):
                 selection.setCurrentIndex(new_index, QtCore.QItemSelectionModel.Current)
 
 
+class UniqueEditableListView(EditableListView):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self._is_drag_drop = False
+
+    def setModel(self, model):
+        current_model = self.model()
+        if current_model:
+            current_model.dataChanged.disconnect(self.on_data_changed)
+        super().setModel(model)
+        model.dataChanged.connect(self.on_data_changed)
+
+    def dropEvent(self, event):
+        self._is_drag_drop = True
+        super().dropEvent(event)
+        self._is_drag_drop = False
+
+    def on_data_changed(self, top_left, bottom_right, roles):
+        # Do not drop duplicates during drag and drop as there is always
+        # a duplicate in the list for a short time in this case.
+        if self._is_drag_drop:
+            return
+        role = roles[0]
+        model = self.model()
+        if role == QtCore.Qt.EditRole:
+            value = model.data(top_left, role)
+            if not value:
+                return
+            # Remove duplicate entries from the model
+            changed_row = top_left.row()
+            row = 0
+            for item in (model.items):
+                if item == value and row != changed_row:
+                    self.model().removeRow(row)
```

Comment:
Is there a reason to not use `model` in place of `self.model()` ?

Reply:
Fixed